build.js

```diff
@@ -174,4 +173,37 @@ StyleDictionary.registerFormat({
   },
 });
 
+const flattenTypographyValues = (obj) => {
+  const items = Object.entries(obj);
+  const expanded = {};
+  items.forEach(([k, v]) => {
+    if (v[k]) return (expanded[k] = v[k]);
+    if (k === "ios") return (expanded.ios = v);
+
+    const innerItems = Object.entries(v);
+
+    innerItems.forEach(([item, v]) => {
+      const variants = Object.entries(v);
+      const firstChild = variants[0][1];
+      if (typeof firstChild === "string" || typeof firstChild === "number") {
+        expanded[item] = v;
+        return;
+      }
+      variants.forEach(([variant, value]) => {
+        expanded[item + "-" + variant] = value;
+      });
+    });
+  });
+  return expanded;
+};
+
+StyleDictionary.registerFormat({
+  name: "muiTypography",
+  formatter: ({ dictionary }) => {
+    const formatted = JSON.parse(formatEntries(dictionary?.tokens["type set"]));
+    const flattenedValues = flattenTypographyValues(formatted);
+    return JSON.stringify(flattenedValues);
```

Comment:
Should there be an error raised if dictionary is not defined? Right now I think you would just return an empty string